src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -241,12 +241,16 @@ public override async ValueTask<ReadResult> ReadAsync(CancellationToken cancella
                 {
                     ClearCancellationToken();
 
-                    if (cancellationToken.IsCancellationRequested)
+                    if (tokenSource.IsCancellationRequested && !cancellationToken.IsCancellationRequested)
```

Comment:
How would `cancellationToken.IsCancellationRequested` be true unless it was passed in being true?

Reply:
IsCancellationRequested changes from false to true when the associated CancellationTokenSource.Cancel is called.